src/main/scala/it/pagopa/interop/catalogprocess/common/readmodel/ReadModelCatalogQueries.scala

```diff
@@ -173,12 +173,12 @@ object ReadModelCatalogQueries extends ReadModelQuery {
       case Seq()      => None
       case attributes =>
         Some(Filters.or {
-          Filters.in("data.descriptors.attributes.certified.id.id", attributes.map(_.toString))
-          Filters.in("data.descriptors.attributes.certified.id.ids", attributes.map(_.toString))
-          Filters.in("data.descriptors.attributes.declared.id.id", attributes.map(_.toString))
-          Filters.in("data.descriptors.attributes.declared.id.ids", attributes.map(_.toString))
-          Filters.in("data.descriptors.attributes.verified.id.id", attributes.map(_.toString))
-          Filters.in("data.descriptors.attributes.verified.id.ids", attributes.map(_.toString))
+          Filters.in("data.descriptors.attributes.certified.0.id", attributes.map(_.toString))
+          Filters.in("data.descriptors.attributes.certified.1.id", attributes.map(_.toString))
+          Filters.in("data.descriptors.attributes.declared.0.id", attributes.map(_.toString))
+          Filters.in("data.descriptors.attributes.declared.1.id", attributes.map(_.toString))
+          Filters.in("data.descriptors.attributes.verified.0.id", attributes.map(_.toString))
+          Filters.in("data.descriptors.attributes.verified.1.id", attributes.map(_.toString))
```

Comment:
Maybe I'm missing something, but this implementation is wrong: it assumes that each kind of attribute has a maximum of 2 attributes.
The structure is an array containing an array, so here we should use `elemMatch`.

Reply:
I agree with @beetlecrunch , this code works only on the first 2 attributes of each type